assets/js/components/Dashboard/PendingScreenDetail.tsx

```diff
@@ -0,0 +1,177 @@
+
+  return [direction, platformLocation].join(" ");
+};
+
+const getScreenLocationDescription = (config: ScreenConfiguration) => {
+  switch (config.app_id) {
+    case "gl_eink_v2":
+      return getGLScreenLocationDescription(config);
+    default:
+      console.warn(
+        `getScreenLocationDescription not implemented for ${config.app_id}`
+      );
+      return "";
+  }
+};
+
+const PendingScreenDetail: ComponentType<Props> = ({
+  screenID,
+  isLive,
+  config,
+  isHiddenOnPlacesPage,
+  onClickHideOnPlacesPage,
+}: Props): JSX.Element => {
+  const dispatch = useScreenplayDispatchContext();
+  const screensUrl = document
+    .querySelector("meta[name=screens-url]")
+    ?.getAttribute("content");
+
+  const queueToastExpiration = () => {
+    setTimeout(
+      () =>
+        dispatch({
+          type: "SHOW_LINK_COPIED",
+          showLinkCopied: false,
+        }),
+      5000
+    );
+  };
+
+  const locationDescription = getScreenLocationDescription(config);
+
+  const fullScreenUrl = isLive
+    ? `${screensUrl}/v2/screen/${screenID}`
+    : `${screensUrl}/v2/screen/pending/${screenID}`;
+
+  // If screen is already live, this matches `fullScreenUrl`.
+  // If screen is pending, this is its future (but not yet working) live url.
+  const liveScreenUrl = `${screensUrl}/v2/screen/${screenID}`;
+
+  return (
+    <div
+      className="pending-screen-detail"
+      onClick={(e: SyntheticEvent) => e.stopPropagation()}
+    >
+      <Container fluid className="h5">
+        <Row>
+          {locationDescription && (
+            <Col className="d-flex justify-content-start location-description">
+              {locationDescription}
+            </Col>
+          )}
+          <Col className="d-flex justify-content-end location-description">
+            {isLive ? (
+              <>
+                <div className="live-icon">
+                  <LightningChargeFill size={16} color="#eff193" />
+                </div>
+                <div className="pending-or-live">Live · Read-only</div>
+              </>
+            ) : (
+              <>
+                <div className="live-icon">
+                  <ClockFill size={16} />
+                </div>
+                <div className="pending-or-live">Pending</div>
+              </>
+            )}
+          </Col>
+        </Row>
+        <Row>
+          <Col>
+            Screen ID:{" "}
+            <span className="screen-id body--regular">{screenID}</span>
+          </Col>
+        </Row>
+        <Row>
+          <Col lg="auto" className="pe-3">
+            <div className="screen-url">Live screen URL:</div>
+            {isLive ? (
+              <a className="url body--regular" href={liveScreenUrl}>
+                {liveScreenUrl}
+              </a>
+            ) : (
+              <span className="url url--inactive">{liveScreenUrl}</span>
+            )}
+            <div className="screen-url-row-button">
+              <OpenInTabButton url={fullScreenUrl} />
```

Comment:
Yeah, Ana wanted this. The button actually does work because for a pending screen, `fullScreenUrl` points to the page to view it full-screen. (See [this route](https://github.com/mbta/screens/blob/main/lib/screens_web/router.ex#L98) and its corresponding controller action in the V2 ScreenController.)

It's the reason why there are separate variables for `fullScreenUrl` and `liveScreenUrl`. Still pretty confusing, though...

Reply:
Ahh I see. I made this comment assuming that the URL used was the same as the text. That makes sense though. Especially if Ana wants it.